packages/@uppy/core/src/BasePlugin.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * Core plugin logic that all plugins share.
+ *
+ * BasePlugin does not contain DOM rendering so it can be used for plugins
+ * without an user interface.
+ *
+ * See `Plugin` for the extended version with Preact rendering for interfaces.
+ *
+ * @param {object} main Uppy core object
+ * @param {object} object with plugin options
+ * @returns {Array|string} files or success/fail message
```

Comment:
the params stuff is not really relevant here. i guess it's copy-paste from the existing plugin class, where it's also nonsensical 😅 
```suggestion
 * BasePlugin does not contain DOM rendering so it can be used for plugins
 * without a user interface.
 *
 * See `Plugin` for the extended version with Preact rendering for interfaces.
```